cmd/autoheal/server.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright (c) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+
+	"github.com/openshift/autoheal/pkg/signals"
+)
+
+// Values of the command line options:
+var (
+	serverKubeAddress string
+	serverKubeConfig  string
+	serverConfigFile  string
+)
+
+var serverCmd = &cobra.Command{
+	Use:   "server",
+	Short: "Starts the auto-heal server",
+	Long:  "Starts the auto-heal server.",
+	Run:   serverRun,
+}
+
+func init() {
+	serverFlags := serverCmd.Flags()
+	serverFlags.StringVar(
+		&serverKubeConfig,
+		"kubeconfig",
+		filepath.Join(homedir.HomeDir(), ".kube", "config"),
+		"Path to a Kubernetes client configuration file. Only required when running "+
+			"outside of a cluster.",
+	)
+	serverFlags.StringVar(
+		&serverKubeAddress,
+		"master",
+		"",
+		"The address of the Kubernetes API server. Overrides any value in the Kubernetes "+
+			"configuration file. Only required when running outside of a cluster.",
+	)
+	serverFlags.StringVar(
+		&serverConfigFile,
+		"config-file",
+		"autoheal.yml",
+		"The location of the configuration file.",
+	)
+}
+
+func serverRun(cmd *cobra.Command, args []string) {
+	var err error
+
+	// Set up signals so we handle the first shutdown signal gracefully:
+	stopCh := signals.SetupSignalHandler()
+
+	// Load the Kubernetes configuration:
+	var config *rest.Config
+	_, err = os.Stat(serverKubeConfig)
+	if os.IsNotExist(err) {
```

Comment:
What if there was another error?

Reply:
You are right @zgalor, that isn't checking all the potential errors. However, I think that it should be fixed in a separate patch. I created #14 to track it.